lib/MT/DataAPI/Callback/TemplateMap.pm

```diff
@@ -86,6 +99,16 @@ sub _retrieve_archive_types {
             next unless $archiver->entry_based;
             next if $archiver->entry_class eq 'page';
         }
+        elsif ( $obj_type eq 'ct_archive' ) {
+
+            # only include if it is NOT a contenttype-based archive type
+            next if $archiver->contenttype_based;
```

Comment:
... and ```next unless $archiver->contenttype_group_based;``` (and maybe ```next if !$has_cat_field && $archiver->contenttype_category_based;``` and/or ```next if !$has_date_field && $archiver->contenttype_date_based```)?

Reply:
And do we need to check ```current_api_version``` as well, because this may be called from v2 endpoints?

Reply:
@charsbar 

> And do we need to check current_api_version as well, because this may be called from v2 endpoints?

No check is needed because $obj_type is always not 'ct' or 'ct_archive' here when called from v2 endpoint.

https://github.com/movabletype/movabletype/blob/a4e53f9a6664763236015715768b7abb76074434/lib/MT/DataAPI/Endpoint/v2/TemplateMap.pm#L109

Reply:
@miniuchi That's true, but IMHO additional api version check here makes it easier to see since when ct/ct_archive are supported, esp. when we revisit the code later.